src/clr-angular/forms/datepicker/providers/date-io.service.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+import {
+    BIG_ENDIAN,
+    DEFAULT_LOCALE_FORMAT,
+    InputDateDisplayFormat,
+    LITTLE_ENDIAN,
+    LITTLE_ENDIAN_REGEX,
+    MIDDLE_ENDIAN,
+    MIDDLE_ENDIAN_REGEX,
+    NON_NEGATIVE_NUMBER,
+    SEPARATORS
+} from "../utils/constants";
+import {getNumberOfDaysInTheMonth, parseToFourDigitYear} from "../utils/date-utils";
+
+import {LocaleHelperService} from "./locale-helper.service";
+
+@Injectable()
+export class DateIOService {
+    public cldrLocaleDateFormat: string = DEFAULT_LOCALE_FORMAT;
+    private localeDisplayFormat: InputDateDisplayFormat = LITTLE_ENDIAN;
+
+    constructor(private _localeHelperService: LocaleHelperService) {
+        this.cldrLocaleDateFormat = this._localeHelperService.localeDateFormat;
+        this.initializeLocaleDisplayFormat();
+    }
+
+    private _inputDate: string;
+
+    get inputDate(): string {
```

Comment:
You alternate between `dateStr` and `inputDate` to denote the stringified date. You should pick one (I have a preference for `DateStr`.)